feat(bindings): expose RMSD to `TM_result` object

cvigilv opened the pull request:
This is a PR draft to expose RMSD (and other results from TMalign, see #33).

cvigilv commented:
I modified the necessary files to expose the RMSD (i think), but I'm unable to build the package. Could you give me a hand on how to achieve this?

cvigilv commented:
Ok, got it working...

```python
>>> import numpy as np
>>> from tmtools import tm_align
>>> coords1 = np.array( [[1.2, 3.4, 1.5], [4.0, 2.8, 3.7], [1.2, 4.2, 4.3], [0.0, 1.0, 2.0]])
>>> coords2 = np.array( [[2.3, 7.4, 1.5], [4.0, 2.9, -1.7], [1.2, 4.2, 4.3]])
>>> seq1 = "AYLP"
>>> seq2 = "ARN"
>>> res = tm_align(coords1, coords2, seq1, seq2)
>>> res.t
array([ 2.94676159,  5.55265245, -1.75151383])
>>> res.u
array([[ 0.40393231,  0.04161396, -0.91384187],
       [-0.59535733,  0.77040999, -0.22807475],
       [ 0.69454181,  0.63618922,  0.33596866]])
>>> res.tm_norm_chain1
0.31058333263221427
>>> res.tm_norm_chain2
0.4141111101762857
>>> res
<tmtools._bindings.TMResult object at 0x102f976b0>
>>> res.
res.rmsd           res.t              res.tm_norm_chain1 res.tm_norm_chain2 res.u
>>> res.rmsd
3.0430316597e-314

```

cvigilv commented:
Interested in exposing more results? I can add some relevant ones such as:

- Aligned sequences (for both sequences)
- Alignment length
- Global TMscore

jvkersch commented on src/_wrapper.cpp:
```diff
@@ -38,7 +38,7 @@ void _tmalign_wrapper(double **xa, double **ya, const char *seqx,
   int L_ali; // Aligned length in standard_TMscore
   double Liden = 0;
-  double TM_ali, rmsd_ali; // TMscore and rmsd in standard_TMscore
+  double TM_ali; // TMscore and rmsd in standard_TMscore
```
```suggestion
  double TM_ali; // TMscore in standard_TMscore
```

jvkersch reviewed (commented):
This looks like the right approach. I would recommend adding a test for RMSD and then we are good to go.

Once the PR is ready for review, could you also add a note `closes #19` to the PR description? Then the merging of your PR will automatically also close the issue.

jvkersch commented:
> Interested in exposing more results? I can add some relevant ones such as:

Yes, but I would do these as separate PRs. Let's keep this one small and self-contained.

jvkersch reviewed (approved):
LGTM! Thanks also for taking the lead on this.

jvkersch commented:
@cvigilv Are you OK with resolving the merge conflict, or do you want me to?

jvkersch reviewed (changes requested):
Actually, just one additional thing to implement: would it be possible to have a test where the RMSD is not equal to 0? Just to avoid that there's something broken internally and we just end up seeing the default 0, not the actual value.

cvigilv commented:
Sure! I'll tackle it tomorrow.

cvigilv reviewed (commented):
👍🏼

cvigilv commented:
Ok, so the implementation was erroneous.

It seems that `rmsd_ali` in the CPP code is not the RMSD after structural alignment. Noticed this whenever I added a test comparing two different structures (expanded on the test set you already had, verifying values by manually runnin TMalign).

Fixed the code to expose the correct RMSD value and added RMSD to the example script and README files so users know that this value is available.

Resolved the conflict to, so it should be ready to go. Let me know if you want to fix the erroneous output variable naming or if its better to do other PR fixing that.

jvkersch reviewed (approved):
LGTM!
